hnn_core/extracellular.py

```diff
@@ -28,6 +28,38 @@
 from .externals.mne import _validate_type, _check_option
 
 
+def _calculate_csd2d(lfp_data, ch_axis=0, delta=1):
+    """current source density (csd) estimation
+    
+    The three-point finite-difference approximation of the
+    second spatial derivative for computing 1-dimensional CSD
+    with border electrode interpolation
+    csd[electrode] = -(LFP[electrode - 1] - 2*LFP[electrode] +
+                       LFP[electrode + 1]) / spacing ** 2
+    
+    Parameters
+    ----------
+    lfp_data : channels x times array
+        LFP data
+    ch_axis : int
+        axis of electrode array, default is zero.
+    delta : int
```

Comment:
Apologies for being late to the review here but the extracellular array doesn't require a fixed spacing between electrodes. This could be pretty quickly resolved by passing the list of positions to the function, taking the `np.diff()` on the positions, and doing an element-wise division 2nd derivative of the LFP data.

Reply:
Fixed spacing is standard for shank electrodes used in experimentation.. would it be reasonable to just make this a requirement?

Reply:
fixed in https://github.com/jonescompneurolab/hnn-core/pull/517/commits/c1375409d707ea7d437034d6f2270ff19ba7f74b

Reply:
It's inferred from electrode position using `_get_laminar_z_coords()`.

Reply:
Sorry, I don't think I answered your question properly. `delta` still needs to be passed into this function since it's inferred from the electrode position alongside the other colinearity + equispacing checks in `_get_laminar_z_coords()`.